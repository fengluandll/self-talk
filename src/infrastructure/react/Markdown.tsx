import React from "react";
import ReactMarkdown from "react-markdown";
import styled, { css } from "styled-components";

const Container = styled.div`
  cursor: text;

  > :first-child {
    margin-top: 0em;
  }

  > :last-child {
    margin-bottom: 0em;
  }

  ${[1, 2, 3, 4, 5, 6].map(
    level => css`
      h${level} {
        font-size: 1em;

        &::before {
          content: "${"#".repeat(level)} ";
        }
      }
    `
  )};

  a {
    color: salmon;

    &:visited {
      color: salmon;
    }
  }

  table {
    border-collapse: collapse;
  }

  table,
  td,
  th {
    border: 1px solid grey;
  }

  td,
  th {
    padding: 0.4em 0.6em;
  }
`;

interface IProps {
  children: string;
}

export const Markdown = ({ children }: IProps) => (
  <Container>
    <ReactMarkdown source={children} />
  </Container>
);
