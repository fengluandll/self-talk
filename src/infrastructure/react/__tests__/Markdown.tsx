import React from "react";
import { create } from "react-test-renderer";
import { Markdown } from "../Markdown";

it("renders", () => {
  expect(create(<Markdown># Foo</Markdown>).toJSON()).toMatchSnapshot();
});
