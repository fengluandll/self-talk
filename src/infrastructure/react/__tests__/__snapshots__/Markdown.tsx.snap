// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa lkTkYw"
>
  <h1>
    Foo
  </h1>
</div>
`;
